code/modules/power/geothermal/_geothermal.dm

```diff
@@ -0,0 +1,103 @@
+var/global/const/GEOTHERMAL_EFFICIENCY_MOD =   0.2
+var/global/const/GEOTHERMAL_PRESSURE_LOSS =    0.3
+var/global/const/MAX_GEOTHERMAL_PRESSURE =    2000
+var/global/const/GEOTHERMAL_PRESSURE_TO_POWER = 100
+
+/obj/machinery/power/geothermal
+	icon = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state = "geothermal-base"
+	var/tmp/neighbors = 0
+	var/tmp/current_pressure = 0
+	var/tmp/processing = FALSE
+	var/efficiency = 0.5
+
+/obj/machinery/power/geothermal/RefreshParts()
+	..()
+	efficiency = Clamp(total_component_rating_of_type(/obj/item/stock_parts/capacitor) * GEOTHERMAL_EFFICIENCY_MOD, GEOTHERMAL_EFFICIENCY_MOD, 1)
+
+/obj/machinery/power/geothermal/proc/add_pressure(var/pressure)
+	current_pressure = Clamp(current_pressure + pressure, 0, MAX_GEOTHERMAL_PRESSURE)
+	if(!processing)
```

Comment:
can just check `is_processing`, it's set to SS when processing and null when not